File: moxy/compiler/src/main/java/com/omegar/mvp/compiler/pipeline/Receiver.java
package com.omegar.mvp.compiler.pipeline;

/**
 * Created by Anton Knyazev on 03.12.2020.
 */
public abstract class Receiver<I> extends Processor<I, Void> {

    @Override
    public void process(I input, PipelineContext<Void> context) {
        receive(input);
    }

    public abstract void receive(I input);

}

File: moxy/compiler/src/main/java/com/omegar/mvp/compiler/pipeline/Receiver.kt
package com.omegar.mvp.compiler.pipeline

/**
 * Created by Anton Knyazev on 03.12.2020.
 */
abstract class Receiver<I> : Processor<I, Void?>() {

    override fun process(input: I, context: PipelineContext<Void?>?) {
        receive(input)
    }

    abstract fun receive(input: I)
}